### PR TITLE
Clone and WithContext #471

### PR DESCRIPTION
**Pull request by smacker**
`client.Copy()` method which will properly set `client.cmdable.process = client.Process`
And `client.WithContext(c)` that will do the same + set context.

### CODE REVIEW

File: redis_go1.7.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// +build go1.7
```
Comment (vmihailenco) [inline redis_go1.7.go]: Add go1.8 too and rename to `redis_context.go`? go1.8 should be released really soon.

File: redis_legacy.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// +build !go1.7
```
Comment (vmihailenco) [inline redis_legacy.go]: Probably rename to `redis_no_context.go`.

File: redis.go [hunk 3]
```diff
@@ -309,6 +301,13 @@ func NewClient(opt *Options) *Client {
 	return newClient(opt, newConnPool(opt))
 }
 
+func (c *Client) Clone() *Client {
```
Comment (vmihailenco) [inline redis.go]: Can we unexport this?

Please rename `Clone` to `Copy`. I think this is more idiomatic name :)

**Comment by vmihailenco** [comment]
Overall looks good. Left some minor comments to consider.

File: redis_go1.7.go (hunk 1 above)
Comment (smacker) [inline redis_go1.7.go]: `"go1.7", from Go version 1.7 onward`
https://golang.org/pkg/go/build/

As I understand this build tag covers all version >=1.7

**Comment by smacker** [comment]
Comments are fixed.

**Comment by vmihailenco** [comment]
Thanks! Merging.

**Pull request merged**